apps/prairielearn/python/prairielearn.py

```diff
@@ -1588,37 +1589,39 @@ def is_correct_ndarray2D_ra(a_sub, a_tru, rtol=1e-5, atol=1e-8):
     return np.allclose(a_sub, a_tru, rtol, atol)
 
 
-def is_correct_scalar_ra(a_sub, a_tru, rtol=1e-5, atol=1e-8):
+def is_correct_scalar_ra(
+    a_sub: ArrayLike, a_tru: ArrayLike, rtol: float = 1e-5, atol: float = 1e-8
+) -> bool:
     """Compare a_sub and a_tru using relative tolerance rtol and absolute tolerance atol."""
-    return np.allclose(a_sub, a_tru, rtol, atol)
+    return bool(np.allclose(a_sub, a_tru, rtol, atol))
 
 
-def is_correct_scalar_dd(a_sub, a_tru, digits=2):
+def is_correct_scalar_dd(a_sub: ArrayLike, a_tru: ArrayLike, digits: int = 2) -> bool:
     """Compare a_sub and a_tru using digits many digits after the decimal place."""
 
     # If answers are complex, check real and imaginary parts separately
     if np.iscomplexobj(a_sub) or np.iscomplexobj(a_tru):
-        return is_correct_scalar_dd(
-            a_sub.real, a_tru.real, digits=digits
-        ) and is_correct_scalar_dd(a_sub.imag, a_tru.imag, digits=digits)
+        real_comp = is_correct_scalar_dd(a_sub.real, a_tru.real, digits=digits)  # type: ignore
+        imag_comp = is_correct_scalar_dd(a_sub.imag, a_tru.imag, digits=digits)  # type: ignore
+        return real_comp and imag_comp
 
     # Get bounds on submitted answer
     eps = 0.51 * (10**-digits)
     lower_bound = a_tru - eps
     upper_bound = a_tru + eps
 
     # Check if submitted answer is in bounds
-    return (a_sub > lower_bound) & (a_sub < upper_bound)
+    return bool((a_sub > lower_bound) & (a_sub < upper_bound))
```

Comment:
Is there a specific reason this is using a bitwise `&` operator instead of the usual `and` operator?

Reply:
I'm not sure; I believe this change was originally pushed by @\eliotwrobson when they added typechecking to the [prairielearn.py](https://github.com/PrairieLearn/PrairieLearn/pull/7934/files/f9ae7ffa00fe0f94b517feaade01ad6e920a6bcd#diff-58735c0379465a7c9e00a320feb090342b1561025053fbe98c118fd44484d375) file. You'll have to ask them.

Reply:
I don't think so, just what was here before (for whatever odd reason).